### PR TITLE
Fix issue #847

### LINKED ISSUE #847: java.nio.BufferUnderflowException
Hello community,
I am not sure if this is a bug or a mistake of mine. I'm trying to upload images from an HTML page, it works quite well with small images (max size = 16kb), but with large images the server throws the exception java.nio.BufferUnderflowException. 

grafik

Is there a way to increase the frame capacity here?
I use the Firefox Quantum as a browser.

### PR DESCRIPTION
**Pull request by marci4**
## Description

This PR fixes the issue introduced with #805 

## Related Issue

Fixes #847 

## Motivation and Context

Bug fix

## How Has This Been Tested?

Added a new test

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by PhilipRoman** [comment]
Can you please explain what's happening in the Issue847Test? Why not simply send a large message and test it was received correctly?

**Comment by marci4** [comment]
> 
> 
> Can you please explain what's happening in the Issue847Test? Why not simply send a large message and test it was received correctly?

I was not able to reproduce with one single tcp packet.
Wireshark also showed that firefox is sending multiple tcp fragements and when recreating this behavior, the issue occured

File: src/main/java/org/java_websocket/drafts/Draft_6455.java [hunk 1]
```diff
@@ -525,7 +525,8 @@ private Framedata translateSingleFrame( ByteBuffer buffer ) throws IncompleteExc
      * @throws LimitExceededException if the payload length is to big
      */
     private TranslatedPayloadMetaData translateSingleFramePayloadLength(ByteBuffer buffer, Opcode optcode, int oldPayloadlength, int maxpacketsize, int oldRealpacketsize) throws InvalidFrameException, IncompleteException, LimitExceededException {
-        int payloadlength = oldPayloadlength, realpacketsize = oldRealpacketsize;
+        int payloadlength = oldPayloadlength,
```
Comment (PhilipRoman) [inline src/main/java/org/java_websocket/drafts/Draft_6455.java]: Are you sure that fixed the code style issue? 🤔 Probably meant to put them in separate declarations

Comment (marci4) [reply src/main/java/org/java_websocket/drafts/Draft_6455.java]: Does the magic ;) (only when we are in a non strict mode according to the documentation)

**Pull request merged**